File: advent-2018/src/main/java/be/robinvercruysse/advent/day1/Day1.java
package be.robinvercruysse.advent.day1;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

public class Day1 {
    public static void main(String[] args) throws IOException {
        InputStream input = Day1.class.getClassLoader().getResourceAsStream("be/robinvercruysse/advent/day1/input");
        BufferedReader reader = new BufferedReader(new InputStreamReader(input));
        String line = null;
        int freq = 0;
        while ((line = reader.readLine()) != null) {
            if (line.startsWith("-")) {
                int value = Integer.parseInt(line.substring(1));
                freq -= value;
            } else if (line.startsWith("+")) {
                int value = Integer.parseInt(line.substring(1));
                freq += value;
            } else {
                System.out.println("anomaly: " + line);
            }
        }
        System.out.println("result: " + freq);
    }
}

File: 2019/day1/Day1.py
with open('input') as fp:
    line = fp.readline()
    total_fuel = 0
    while line:
        mass = int(line)
        fuel = int(mass / 3) - 2
        total_fuel = total_fuel + fuel
        while fuel > 0:
            mass = fuel
            fuel = int(mass / 3) - 2
            if fuel < 0:
                fuel = 0
            total_fuel = total_fuel + fuel
        line = fp.readline()
    print(str(total_fuel))
